azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/ImplementationInstance.kt

```diff
@@ -163,9 +163,19 @@ class JarImplementationInstance<
             var result: TaskResult<S, R>? = null
 
             override fun run() {
-                val implementation = Implementation.locateImplementations<AF, CF, QF, AGF>().single()
-                val implementationTask = TaskWrapper(taskType, implementation, task)
-                result = implementationTask.run(scenario)
+                val implementations = Implementation.locateImplementations<AF, CF, QF, AGF>()
+                result = if (implementations.size == 1) {
+                    val implementation = implementations.single()
+                    Log.debug("Running task type=$taskType implementation=${implementation.name}")
+                    val implementationTask = TaskWrapper(taskType, implementation, task)
+                    implementationTask.run(scenario)
+                } else {
+                    TaskResult(taskType,
+                        "Unknown",
+                        null,
+                        exception = IllegalStateException("No implementations found"),
```

Comment:
I know it shouldn't be possible, but it makes me a little uneasy that this message would be shown even when more than one implementation is found.

Reply:
I'll change it as I'm still confused why this PR is showing an out-of-date sha anyway!